src/SFA.DAS.ApplyService.Web/Views/Roatp/ApplicationUnsuccessfulPostGateway.cshtml

```diff
@@ -0,0 +1,247 @@
+            </p>
+
+
+            <h1 class="govuk-heading-l">Why your application was unsuccessful</h1>
+            <p class="govuk-body">
+                Your application was unsuccessful as the answer for the following questions are insufficient
+            </p>
+            @if (Model.Sequences != null)
+            {
+                @foreach (var sequence in Model.Sequences)
+                {
+                    <h1 class="govuk-heading-l">@sequence.SequenceTitle</h1>
+                    foreach (var seqSection in sequence.Sections)
+                    {
+
+                        var inSectors = sequence.SequenceNumber == RoatpWorkflowSequenceIds.DeliveringApprenticeshipTraining
+                                        && seqSection.SectionNumber == RoatpWorkflowSectionIds.DeliveringApprenticeshipTraining.YourSectorsAndEmployees;
+
+                        if (inSectors)
+                        {
+                            <h2 class="govuk-heading-m">Sectors</h2>
+                            <ul class="govuk-list govuk-list--bullet">
+                                @foreach (var sectionPage in seqSection.Pages)
+                                {
+
+                                    <li> @sectionPage.Title - PENDING</li>
+
+                                }
+                            </ul>
+
+                        }
+                        else
+                        {
+                            @foreach (var sectionPage in seqSection.Pages)
+                            {
+                                var isTabularData = false;
+                                @foreach (var dummy in from question in sectionPage.Questions
+                                    from answers
+                                        in sectionPage.PageOfAnswers
+                                    from answer
+                                        in answers.Answers.Where(x => x.QuestionId == question.QuestionId)
+                                    where question.Input.Type == QuestionType.TabularData
+                                    select question)
+                                {
+                                    isTabularData = true;
+                                }
+
+
+                                if (isTabularData)
+                                {
+                                    <h2 class="govuk-heading-m"> @sectionPage.Title</h2>
+
+                                    @foreach (var question in sectionPage.Questions)
+                                    {
+                                        @foreach (var answers in sectionPage.PageOfAnswers)
+                                        {
+                                            foreach (var answer in answers.Answers.Where(x => x.QuestionId == question.QuestionId))
```

Comment:
I'd use LINQ to flatten then query. That way you'll be at something like nLog(n) rather than n^2

Reply:
I did some, but the flattening produced some unreadable stuff in places, and I left it as is in those.